### PR TITLE
General cleanup in source/net

### PR DESCRIPTION
**Pull request by marcusnaslund**
Updated to conform with new versions of rock and style guides for `magic`, remove gcc warnings, and also includes some bug fixes.

Added a blank test which makes sure the code in `source/net` gets processed when running `./test.sh`. There is already an issue on writing actual tests: #757

### CODE REVIEW

File: source/net/ServerSocket.ooc [hunk 1]
```diff
@@ -118,22 +114,22 @@ ServerSocket: class extends Socket {
...
 	*/
-	accept: func ~withClosure (f: func (TCPServerReaderWriterPair) -> Bool) {
-		if (!listening)
-			listen()
+	accept: func ~withClosure (f: Func (TCPServerReaderWriterPair) -> Bool) {
+		if (!this listening)
+			this listen()
 
-		loop(||
+		while (true) {
```
Comment (marcusnaslund) [inline source/net/ServerSocket.ooc]: I don't know if `loop` was a keyword that disappeared or what was going on here?

File: source/net/translation.ooc [hunk 2]
```diff
@@ -10,63 +10,66 @@ import berkeley, Socket, Address
...
+		//version (!windows) {
 			// use built-in!
-			return inet_ntop(addressFamily, address, destination, destinationSize)
-		}
+			// TODO: Using the built-in inet_ntop results in a gcc warning because inet_ntop returns a const char*,
+			// and the const qualifier is not supported by rock...
+			//result = inet_ntop(addressFamily, address, destination, destinationSize)
+		//} else {
+			// Roll our own implementation, based on:
+			// https://github.com/pkulchenko/luasocket/blob/5a58786a39bbef7ed4805821cc921f1d40f12068/src/inet.c#L512
```
Comment (marcusnaslund) [inline source/net/translation.ooc]: As the comment says, `inet_ntop` returns `const char*`, but the closest thing we have is `CString` which is a cover of `char*`...

File: source/net/ServerSocket.ooc (hunk 1 above)
Comment (thomasfanell) [inline source/net/ServerSocket.ooc]: https://github.com/fasterthanlime/rock/blob/master/sdk/lang/Abstractions.ooc

Comment (marcusnaslund) [inline source/net/ServerSocket.ooc]: Oh yes, I remember that file. Thanks.

**Comment by marcusnaslund** [comment]
Who feels like doing some networking peer review? @jonathanudd maybe?

File: source/net/Address.ooc [hunk 3]
```diff
@@ -172,155 +172,136 @@ IP4Address: class extends IPAddress {
 			InvalidAddress new("Could not parse address") throw()
 		}
```
Comment (jonathanudd) [inline source/net/Address.ooc]: Brackets could be removed here and on the if above.

File: source/net/Address.ooc [hunk 4]
```diff
@@ -172,155 +172,136 @@ IP4Address: class extends IPAddress {
...
 	}
-
 	init: func ~wildcard {
```
Comment (jonathanudd) [inline source/net/Address.ooc]: Do we want this kind of functions on one line?

Comment (thomasfanell) [inline source/net/Address.ooc]: Only if the nearest neighbours also can be compacted to one line.

File: source/net/Address.ooc [hunk 5]
```diff
@@ -350,39 +331,34 @@ SocketAddress: abstract class {
...
 		"[%s]:%d" format(host() toString() toCString(), port())
 	}
 	new: static func (host: IPAddress, nPort: Int) -> This {
+		result: This
 		if (host family == AddressFamily IP4) {
 			ip4Host := host as IP4Address
-			return SocketAddressIP4 new(ip4Host ai, nPort)
+			result = SocketAddressIP4 new(ip4Host ai, nPort)
 		}
 		else if (host family == AddressFamily IP6) {
```
Comment (jonathanudd) [inline source/net/Address.ooc]: The `else if` could be moved to the line above. The same applies for the else below.

**Comment by jonathanudd** [comment]
There are some functions that could be converted to one liners but I don't know if that will increase readability.
Apart from the comments this looks good.

**Comment by marcusnaslund** [comment]
Thank you, updated according to comments. Just waiting on PRInt to give a green flag.

**Pull request merged**